docs/developer-guide.md

```diff
@@ -12,55 +14,89 @@
 
 ## Setting up a dev environment
 
-Setting up a dev environment should be as simple as running the commands below in your terminal:
+Setting up a dev environment should be as simple as running the commands
+below in your terminal:
 ```
 git clone https://github.com/zulip/zulip-mobile.git
 cd zulip-mobile
 npm install
 ```
 
-Unlike the [Zulip](https://github.com/zulip/zulip) server project, we use the host machine directly for development instead of provisioning a VM.
+Unlike the [Zulip](https://github.com/zulip/zulip) server project, we use
+the host machine directly for development instead of provisioning a VM.
 
-You'll probably also want to install and provision a [Zulip dev VM](https://zulip.readthedocs.io/en/latest/dev-overview.html) to use for testing.
+You'll probably also want to install and provision a [Zulip dev VM](https://zulip.readthedocs.io/en/latest/dev-overview.html) to use for
+testing.
 
 
 ## Running on iOS simulator
-`react-native run-ios` will launch a new terminal with the React Native packager and open up the app in the iOS simulator.
+`react-native run-ios` will launch a new terminal with the React Native
+packager and open up the app in the iOS simulator.
 
-It will also launch a browser tab in Chrome with the React Native debugger. `console.log` statements in React Native will end up in the JS console on this tab.
+It will also launch a browser tab in Chrome with the React Native debugger.
+`console.log` statements in React Native will end up in the JS console on
+this tab.
 
 ## Running on an iOS device
 1. [Set up the dev environment](#setting-up-a-dev-environment)
 2. Connect your iOS device
 3. Within the repo, `$ open ios/ZulipMobile.xcodeproj/` to open Xcode
-4. Select your device as the `build target` (from [this guide](https://facebook.github.io/react-native/docs/running-on-device-ios.html))
-5. Hit the `build and run` button
+4. Change BundleIdentifier for both ZulipMobile and ZulipMobileTests to a
+unique string, e.g. `<username>ZulipMobile`
+5. Select your device as the `build target` (from [this guide](https://facebook.github.io/react-native/docs/running-on-device-ios.html))
+6. Hit the `build and run` button (make sure your device is unlocked)
+7. If it's the first time you're running the app, you need to trust the
+developer and the app in `Settings > General > Device Management > Developer
+App` - make sure you are connected to WiFi, as it often doesn't work with
+mobile networks
+
+If it's your first time installing an app on an iOS device, you need to
+obtain Apple Developer credentials that will allow you to sign the app.
+Register at https://developer.apple.com. Then use your Apple ID in Xcode
+and choose it as your `Signing > Team` for both ZulipMobile and ZulipMobileTests.
```

Comment:
Maybe we should add something about not committing these changes to `project.pbxproj` to diffs?